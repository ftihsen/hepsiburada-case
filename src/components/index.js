import Header from './Header/Header';
import SubHeader from './Header/SubHeader';
import Card from './Card/Card';
import CardImage from './Card/CardImage';
import CardProductTitle from './Card/CardProductTitle';
import Button from './Button/Button';
import Badge from './Badge/Badge';
import SearchInput from './Search/SearchInput';
import BasketModal from './Basket/BasketModal';
import BasketModalCard from './Basket/BasketModalCard';
import SideBar from './SideBar/SideBar';
import CardProductFilters from './Card/CardProductFilters';
import CardProductPrice from './Card/CardProductPrice';
import SelectBox from './SelectBox/SelectBox';
import Modal from './Modal/Modal';
import ProducList from './ProducList/ProducList';
import Logo from './Logo/Logo';

export {
  Header,
  SubHeader,
  Card,
  CardProductTitle,
  BasketModalCard,
  Button,
  Badge,
  SearchInput,
  BasketModal,
  CardImage,
  SideBar,
  CardProductFilters,
  CardProductPrice,
  SelectBox,
  Modal,
  ProducList,
  Logo,
};
