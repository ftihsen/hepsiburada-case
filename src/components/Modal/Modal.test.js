import { fireEvent, render } from '@testing-library/react';
import { Modal } from 'components';

export const createDocumentListenersMock = () => {
  const listeners = {};
  const handler = (domEl, event) => listeners?.[event]?.({ target: domEl });
  document.addEventListener = jest.fn((event, cb) => {
    listeners[event] = cb;
  });
  document.removeEventListener = jest.fn((event) => {
    delete listeners[event];
  });
  return {
    mouseDown: (domEl) => handler(domEl, 'mousedown'),
    click: (domEl) => handler(domEl, 'click'),
  };
};
describe('Modal Components', () => {
  it('should have the passed class', () => {
    const { container } = render(<Modal open />);

    expect(container.firstChild).toHaveClass('modal-open');
  });

  it('should increment number on click', () => {
    let number = 1;
    const { getByText } = render(<Modal onClickSuccess={() => (number += 1)} />);

    fireEvent.click(getByText('Evet'), new MouseEvent('click'));

    expect(number).toBe(2);
  });

  it('should call onClick() when clicking outside', () => {
    const fireEvents = createDocumentListenersMock();
    const onClick = jest.fn();
    render(
      <Modal onDismiss={onClick}>
        <span>Modal</span>
      </Modal>,
    );

    fireEvents.mouseDown(document.body);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const { container } = render(
      <Modal>
        <span>Modal</span>
      </Modal>,
    );
    expect(container).toMatchSnapshot();
  });
});
