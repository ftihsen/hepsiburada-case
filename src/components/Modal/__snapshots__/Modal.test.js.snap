// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Components should match snapshot 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <p
          class="modal-header-text"
        />
      </div>
      <div
        class="modal-body"
      >
        <span>
          Modal
        </span>
      </div>
      <div
        class="modal-footer"
      >
        <button
          class="button button-small success"
        >
          Evet
        </button>
        <button
          class="button button-small error"
        >
          HayÄ±r
        </button>
      </div>
    </div>
  </div>
</div>
`;
